edx_notifications/data.py

```diff
@@ -0,0 +1,172 @@
+"""
+All pure data objects that the various Data Providers will product. This is help avoid
+generic dictionaries from being passed around, plus this will help avoid any
+implicit database-specific bindings that come with any uses of ORMs.
+"""
+
+from django.core.exceptions import ValidationError
+
+from edx_notifications import const
+
+from edx_notifications.base_data import (
+    StringField,
+    DictField,
+    DateTimeField,
+    IntegerField,
+    EnumField,
+    RelatedObjectField,
+    BaseDataObject
+)
+
+
+class NotificationChannel(BaseDataObject):
+    """
+    Specifies a channel through which a notification is delivered
+    """
+
+    # the internal name of the channel
+    name = StringField()
+
+    # the human readible name of the channel
+    display_name = StringField()
+
+    # the human readible description of the channel
+    description_name = StringField()
```

Comment:
Likewise, do these need to be stored in a database? Or are they just configured and resident in memory?